internal/compiler/generator/cpp.rs

```diff
@@ -82,17 +85,96 @@ mod cpp_ast {
     ///A full C++ file
     #[derive(Default, Debug)]
     pub struct File {
+        pub is_cpp_file: bool,
         pub includes: Vec<String>,
         pub after_includes: String,
         pub namespace: Option<String>,
         pub declarations: Vec<Declaration>,
+        pub resources: Vec<Declaration>, // For now inline, maybe #embed in the future
```

Comment:
```suggestion
        pub resources: Vec<Declaration>,
```

I think the comment is confusing. they should not be `inline` if we split them in a translation unit.
They should be forward declared.

Reply:
Okay :)